### PR TITLE
backport lchown patch to PMIx 3.1.x + 3.2.x

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen2

File: easybuild/easyconfigs/p/PMIx/PMIx-3.2.3-GCCcore-10.3.0.eb [hunk 1]
```diff
@@ -24,7 +24,11 @@ toolchainopts = {'pic': True}
 
 source_urls = ['https://github.com/openpmix/openpmix/releases/download/v%(version)s']
 sources = ['%(namelower)s-%(version)s.tar.bz2']
-checksums = ['9b835f23c2f94a193c14012ee68b3657a61c568598cdd1212a3716b32d41a135']
+patches = ['PMIx-3.1.x_lchown.patch']
+checksums = [
+    {'pmix-3.2.3.tar.bz2': '9b835f23c2f94a193c14012ee68b3657a61c568598cdd1212a3716b32d41a135'},
+    {'PMIx-3.1.x_lchown.patch': '9a4d5e3cb0e619f2e9ac2ffad0b8cb2b22f5c67973080724c79519fb34d55776'},
```
Comment (branfosj) [inline easybuild/easyconfigs/p/PMIx/PMIx-3.2.3-GCCcore-10.3.0.eb:30]:
```
$ grep "chown" -r pmix-3.2.3
pmix-3.2.3/config/install-sh:chownprog=${CHOWNPROG-chown}
pmix-3.2.3/config/install-sh:chowncmd=
pmix-3.2.3/config/install-sh:  -o USER       $chownprog installed files to USER.
pmix-3.2.3/config/install-sh:    -o) chowncmd="$chownprog $2"
pmix-3.2.3/config/install-sh:    # Don't chown directories that already exist.
pmix-3.2.3/config/install-sh:      chowncmd=""
pmix-3.2.3/config/install-sh:    { test -z "$chowncmd" || $doit $chowncmd "$dst"; } &&
pmix-3.2.3/config/install-sh:    { test "$obsolete_mkdir_used$chowncmd$chgrpcmd" = false ||
pmix-3.2.3/config/install-sh:    { test -z "$chowncmd" || $doit $chowncmd "$dsttmp"; } &&
pmix-3.2.3/src/util/pmix_pty.c:    chown(pts_name, getuid(), gid);
pmix-3.2.3/src/mca/ptl/usock/ptl_usock_component.c:    /* chown as required */
pmix-3.2.3/src/mca/ptl/usock/ptl_usock_component.c:        if (0 != chown(address->sun_path, lt->owner, -1)) {
pmix-3.2.3/src/mca/ptl/usock/ptl_usock_component.c:        if (0 != chown(address->sun_path, -1, lt->group)) {
pmix-3.2.3/src/mca/gds/ds12/gds_ds12_lock_pthread.c:            if (0 > chown(lock_ctx->lockfile, (uid_t) uid, (gid_t) -1)){
pmix-3.2.3/src/mca/gds/ds12/gds_ds12_lock_fcntl.c:            if (0 > chown(lock_ctx->lockfile, uid, (gid_t) -1)) {
pmix-3.2.3/src/mca/common/dstore/dstore_segment.c:                if (0 > chown(file_name, (uid_t) uid, (gid_t) -1)){
pmix-3.2.3/src/mca/common/dstore/dstore_segment.c:            if (0 > chown(file_name, (uid_t) uid, (gid_t) -1)){
pmix-3.2.3/src/mca/common/dstore/dstore_base.c:            if (0 > chown(s->nspace_path, (uid_t) s->jobuid, (gid_t) -1)){
pmix-3.2.3/src/mca/common/dstore/dstore_base.c:            if (chown(ds_ctx->base_path, (uid_t) ds_ctx->jobuid, (gid_t) -1) < 0){
```

The use in `pmix-3.2.3/src/util/pmix_pty.c` is not being patched. This is in PMIx 3.2.3, but not in PMIx 3.1.5
```
pmix-3.2.3/src/util/pmix_pty.c:    chown(pts_name, getuid(), gid);
```

**Review by branfosj — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/p/PMIx/PMIx-3.2.3-GCCcore-10.3.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/p/PMIx/PMIx-3.2.3-GCCcore-10.3.0.eb]: fixed in b12d202

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (4 easyconfigs in total)
node3171.skitty.os - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/boegel/b38766e33ef7c129eb7834733ee64db0 for a full test report.

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 5 out of 5 (4 easyconfigs in total)
bear-pg0211u03a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/branfosj/90b3da103f473724b89bc3294b104d3a for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**